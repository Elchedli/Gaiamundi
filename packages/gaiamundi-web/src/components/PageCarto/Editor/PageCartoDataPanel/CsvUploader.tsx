import { useState } from 'react';
import { useMutation } from 'react-query';

import {
  DocumentChartBarIcon,
  ExclamationCircleIcon,
} from '@heroicons/react/24/solid';

import { Alert } from 'components/Alert/Alert';
import { ApiErrorAlert } from 'components/Alert/ApiErrorMessage';
import { Button } from 'components/Button/Button';
import LoadingSpinner from 'components/Icons/LoadingSpinner';
import DropZone from 'components/Inputs/DropZone';
import { useToast } from 'hooks/useToast';
import { ApiError } from 'interfaces/api';
import { Column } from 'interfaces/column';
import { UploadedFile } from 'interfaces/file';
import { uploadCsv } from 'services/page-carto';
import { parseCsvColumns, validateCsv } from 'utils/file';

type CsvUploaderProps = {
  onUpload: (file: UploadedFile) => void;
  onParse?: (columnNames: Column[]) => void;
  onCancel?: () => void;
  onChange: (fileId: number) => void;
};

export const CsvUploader: React.FC<CsvUploaderProps> = ({
  onUpload,
  onParse,
  onCancel,
  onChange,
}) => {
  const { addToast } = useToast();
  const [file, setFile] = useState<UploadedFile | undefined>(undefined);
  const [errors, setErrors] = useState<string>('');

  const { mutateAsync, isError, isLoading, error } = useMutation({
    mutationFn: async (data: { file: File }) => {
      return await uploadCsv(data.file);
    },
    onSuccess: (data: UploadedFile) => {
      setFile(data);
      onUpload(data);
      onChange(data.id);
      addToast({
        title: `Fichier téléchargé`,
        description: `Fichier ${data.name} téléchargé avec succès`,
        type: 'success',
      });
    },
  });

  const handleUpload = async (csvFile: File) => {
    setErrors('');
    try {
      const data = await validateCsv(csvFile);
      if (onParse) {
        const columns = parseCsvColumns(data.meta.fields || []);
        onParse(columns);
      }

      if (Array.isArray(data.errors) && data.errors.length > 0) {
        // eslint-disable-next-line no-console
        console.warn(`CSV Parse errors : `, data.errors);
      }
      // This is done to set the displayed filename after the file is uploaded
      if (data.errors.length > 0) {
        setErrors(
          `Le fichier CSV contient des erreurs dans les lignes : ${data.errors
            .map((e) => e.row)
            .join(', ')}`
        );
      }
      mutateAsync({ file: csvFile });
    } catch (e) {
      addToast({
        title: `Erreur lors de la validation du fichier CSV`,
        description: (e as Error).message,
        type: 'error',
      });
    }
  };

  const handleCancel = () => {
    setFile(undefined);
    setErrors('');
    onCancel && onCancel();
  };

  if (isLoading) {
    return <LoadingSpinner />;
  }

  if (isError) {
    <ApiErrorAlert error={error as ApiError} />;
  }

  return (
    <div className="self-center">
      {file ? (
        <div className="flex flex-col items-center justify-center pt-5 pb-6">
          <DocumentChartBarIcon width={64} height={64} />
          <span className="mb-2 text-sm text-gray-500 dark:text-gray-400">
            {file.name} ({file.mime})
          </span>
          <Button color="red" onClick={handleCancel} size="xs">
            Annuler
          </Button>
        </div>
      ) : (
        <DropZone onUpload={handleUpload} />
      )}
      {errors && (
        <Alert className="p-2 text-sm" icon={ExclamationCircleIcon}>
          {errors}
        </Alert>
      )}
    </div>
  );
};
