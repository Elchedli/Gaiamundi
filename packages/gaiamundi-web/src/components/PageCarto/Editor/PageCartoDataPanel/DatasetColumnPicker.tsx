import { MagnifyingGlassIcon } from '@heroicons/react/24/solid';
import { Radio } from 'components/Inputs/Radio';
import { TextInput } from 'components/Inputs/TextInput';
import { Column } from 'interfaces/column';
import React, { useEffect, useMemo, useState } from 'react';
import DataGrid, {
  Column as DataGridColumn,
  SelectColumn,
} from 'react-data-grid';

type DatasetColumnPickerProps = {
  data: Column[];
  onChange: (selectedColumns: Column[]) => void;
};

const DatasetColumnPicker = React.forwardRef<
  HTMLDivElement,
  DatasetColumnPickerProps
>(({ data, onChange }, ref) => {
  const [selectedColumns, setSelectedColumns] = useState<ReadonlySet<string>>(
    () => new Set<string>()
  );
  const [selectedGeoCode, setSelectedGeoCode] = useState<string | null>(null);
  const [searchKeywords, setSearchKeywords] = useState<string>('');

  const filteredRows = useMemo(() => {
    return searchKeywords
      ? data.filter((row) => {
          return row.name.includes(searchKeywords);
        })
      : data;
  }, [data, searchKeywords]);

  const handleRowSelect = (selection: Set<string>) => {
    setSelectedColumns(selection);
    if (selectedGeoCode && !selection.has(selectedGeoCode)) {
      setSelectedGeoCode(null);
    }
  };

  const handleGeoCodeSelect = (selected: string) => {
    setSelectedGeoCode(selected);
  };

  useEffect(() => {
    const selected = Array.from(selectedColumns)
      .map((name) => {
        return data.find((row) => row.name === name);
      })
      .map(
        (column) =>
          ({
            ...column,
            isGeoCode: column?.name === selectedGeoCode,
          } as Column)
      );
    onChange(selected);
  }, [data, onChange, selectedColumns, selectedGeoCode]);

  const columns: DataGridColumn<Column>[] = [
    SelectColumn,
    {
      key: 'isGeoCode',
      name: 'GéoCode',
      width: 80,
      minWidth: 80,
      maxWidth: 80,
      formatter({ row }) {
        return selectedColumns.has(row.name) ? (
          <Radio
            name="isGeoCode"
            value={row.name}
            checked={row.name === selectedGeoCode}
            onChange={() => handleGeoCodeSelect(row.name)}
          />
        ) : null;
      },
      cellClass: 'text-center',
      frozen: true,
    },
    {
      key: 'name',
      name: 'Colonne',
    },
    {
      key: 'sample',
      name: 'Échantillon',
    },
    {
      key: 'source',
      name: 'Source',
    },
    {
      key: 'validity',
      name: 'Validité',
    },
  ];

  return (
    <div className="border-slate-300" ref={ref}>
      <TextInput
        id="keywords"
        className="w-full mb-2"
        icon={MagnifyingGlassIcon}
        value={searchKeywords}
        onChange={(e) => setSearchKeywords(e.target.value)}
        autoComplete={'false'}
      />
      <DataGrid
        defaultColumnOptions={{
          sortable: false,
          resizable: true,
        }}
        className="fill-grid"
        rowKeyGetter={({ name }) => name}
        columns={columns}
        rows={filteredRows}
        selectedRows={selectedColumns}
        onSelectedRowsChange={handleRowSelect}
      />
    </div>
  );
});

DatasetColumnPicker.displayName = 'DatasetColumnPicker';

export default DatasetColumnPicker;
