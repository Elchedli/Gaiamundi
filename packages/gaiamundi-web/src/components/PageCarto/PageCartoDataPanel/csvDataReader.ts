import { validateCsv } from 'utils/file';

export const readCsvDatas = async (csvFile: File) => {
  try {
    const data = await validateCsv(csvFile);
    if (data.errors.length > 0) {
      console.warn(`CSV Collection parse error : `, data.errors);
    }
    // This is done to set the displayed filename after the file is uploaded
  } catch (e) {
    console.log(e);
  }
};
