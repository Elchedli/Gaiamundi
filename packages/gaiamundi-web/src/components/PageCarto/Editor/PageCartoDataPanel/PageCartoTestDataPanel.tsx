import { mockDataFragmentsApiCollection } from 'utils/mocks/data';
import { PageCartoPanelData } from './PageCartoDataPanel';

export const PageCartoTestDataPanel = () => {
  return (
    <PageCartoPanelData
      dataFragments={mockDataFragmentsApiCollection}
      pageCartoId={286}
    />
  );
};
