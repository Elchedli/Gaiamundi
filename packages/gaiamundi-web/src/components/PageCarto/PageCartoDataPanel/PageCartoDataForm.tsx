import { ExclamationTriangleIcon } from '@heroicons/react/24/solid';
import { UploadedFile } from 'interfaces/file';
import { FC, useState } from 'react';
import { Controller, useForm } from 'react-hook-form';
import { useMutation, useQueryClient } from 'react-query';

import { Alert } from 'components/Alert/Alert';
import { ApiErrorAlert } from 'components/Alert/ApiErrorMessage';
import { Button } from 'components/Button/Button';
import { Label } from 'components/Forms/Inputs/Label';
import { TextInput } from 'components/Forms/Inputs/TextInput';
import { ToggleSwitch } from 'components/Forms/Inputs/ToggleSwitch';
import { useRequireAuth } from 'hooks/useRequireAuth';
import { useToast } from 'hooks/useToast';
import { ApiError } from 'interfaces/api';
import { Column } from 'interfaces/column';
import { DatasetStub } from 'interfaces/dataset';
import { addDataToPageCarto } from 'services/dataset';
import { CsvUploader } from './CsvUploader';
import { DatasetColumnPicker } from './DatasetColumnPicker';

type FormData = DatasetStub & {
  fragmentName: string;
  columns: Column[];
};

type PageCartoDataFormProps = {
  pageCartoId: number;
  onSubmit: (dataForm: FormData) => void;
};

export const PageCartoDataForm: FC<PageCartoDataFormProps> = ({
  pageCartoId,
  onSubmit,
}) => {
  const queryClient = useQueryClient();
  const { user } = useRequireAuth();
  const [columns, setColumns] = useState<Column[]>([]);
  const { addToast } = useToast();
  const {
    register,
    handleSubmit,
    formState: { errors, isValid },
    setValue,
    control,
  } = useForm<FormData>();

  const { isError, error, isLoading, mutateAsync } = useMutation({
    mutationFn: async ({ columns, fragmentName, ...dataset }: FormData) => {
      return await addDataToPageCarto(
        pageCartoId,
        fragmentName,
        { ...dataset, owner: user?.id || 0 },
        columns
      );
    },
    onSuccess: (_response, formData) => {
      addToast({
        title: 'Jeu de données importé',
        description: `Votre jeu de données a été crée avec succès`,
        type: 'success',
      });
      queryClient.invalidateQueries({ queryKey: ['page-carto', pageCartoId] });
      onSubmit(formData);
    },
  });

  const handleFileUpload = (file: UploadedFile) => {
    setValue('name', file?.name || '');
    setValue('fragmentName', file?.name || '');
  };

  const handleFileCancel = () => {
    setValue('name', '');
    setValue('fragmentName', '');
    setColumns([]);
  };

  const handleFileParse = (columns: Column[]) => {
    setColumns(columns);
  };

  const onSubmitForm = (data: FormData) => {
    mutateAsync(data);
  };

  return (
    <div className="p-4 bg-slate-100 border border-slate-300 rounded shadow-xl">
      <form onSubmit={handleSubmit(onSubmitForm)}>
        {isError && <ApiErrorAlert error={error as ApiError} />}
        <div className="flex flex-row">
          <div className="flex flex-col w-1/2 p-2">
            <div className="mt-3">
              <Label htmlFor="name">
                1. Uploader des données à partir d&apos;un fichier CSV
              </Label>
              <Controller
                name="csv"
                control={control}
                defaultValue={undefined}
                rules={{
                  required: 'Le fichier CSV est obligatoire',
                }}
                render={({ field }) => {
                  return (
                    <CsvUploader
                      onUpload={handleFileUpload}
                      onParse={handleFileParse}
                      onCancel={handleFileCancel}
                      {...field}
                    />
                  );
                }}
              />
              {errors.csv && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.csv.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <Label>2. Sélectionnez les colonnes à utiliser :</Label>
              <Controller
                name="columns"
                control={control}
                defaultValue={undefined}
                rules={{
                  required:
                    'Vous devez sélectionner les colonnes ainsi que le Géo Code.',
                  validate: (cols) => {
                    return cols.length > 0 && cols.some((c) => c.isGeoCode);
                  },
                }}
                render={({ field }) => {
                  return <DatasetColumnPicker data={columns} {...field} />;
                }}
              />
            </div>
          </div>
          <div className="flex flex-col w-1/2 p-2">
            <div className="mt-3">
              <Label htmlFor="name">Nom du jeu de données</Label>
              <TextInput
                id="fragmentName"
                className="w-full"
                {...register('fragmentName', {
                  required: `Veuillez saisir le nom de votre jeu de données`,
                })}
              />
              {errors.fragmentName && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.fragmentName.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <Label htmlFor="origin">Origine du fichier</Label>
              <TextInput
                id="origin"
                className="w-full"
                {...register('origin', {
                  required: `Veuillez saisir l'origine de votre fichier de données`,
                })}
              />
              {errors.origin && (
                <div className="mt-2 text-xs text-red-600">
                  {errors.origin.message}
                </div>
              )}
            </div>
            <div className="mt-3">
              <ToggleSwitch
                label={'Accès Public'}
                defaultChecked={false}
                onChange={(v) => {
                  setValue('isPublic', v);
                }}
              />
              <Alert
                className="p-2 text-xs"
                type="warning"
                icon={ExclamationTriangleIcon}
              >
                Attention: Risque de non conformité <br />
                <small>
                  (Origine du fichier, source et validité des colonnes)
                </small>
              </Alert>
            </div>
            <div className="flex justify-end mt-3">
              <Button
                type="submit"
                size="lg"
                isLoading={isLoading}
                disabled={!isValid}
              >
                Valider
              </Button>
            </div>
          </div>
        </div>
      </form>
    </div>
  );
};
