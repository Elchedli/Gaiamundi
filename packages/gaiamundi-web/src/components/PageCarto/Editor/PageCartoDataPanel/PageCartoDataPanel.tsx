import { PlusIcon } from '@heroicons/react/24/solid';
import { Alert } from 'components/Alert/Alert';
import { Button } from 'components/Button/Button';
import { useModal } from 'hooks/useModal';
import { ApiCollection } from 'interfaces/api';
import { Column } from 'interfaces/column';
import { DataFragment } from 'interfaces/data-fragment';
import { FC, useMemo } from 'react';
import DataGrid from 'react-data-grid';
import { PageCartoDataForm } from './PageCartoDataForm';

type PageCartoPanelDataProps = {
  dataFragments: ApiCollection<DataFragment>;
  pageCartoId: number;
  isTested?: boolean;
};

export const PageCartoPanelData: FC<PageCartoPanelDataProps> = ({
  dataFragments,
  pageCartoId,
  isTested = false,
}) => {
  const { showModal, hideModal } = useModal();

  const fragments = dataFragments?.data || [];
  const rows = useMemo(
    () =>
      fragments.reduce((acc, curr) => {
        const cols = curr.attributes.columns
          .filter((column) => {
            return !column.isGeoCode;
          })
          .map((column) => {
            return {
              ...column,
              dataset: curr.attributes.dataset.data.attributes.name,
            };
          });
        return acc.concat(cols);
      }, [] as (Column & { dataset: string })[]),
    [dataFragments]
  );
  return (
    <div>
      <div className="mt-5 text-right">
        <Button
          icon={PlusIcon}
          onClick={() =>
            showModal({
              title: 'Importer un jeu de données',
              Component: PageCartoDataForm,
              props: { pageCartoId, onSubmit: hideModal },
            })
          }
        >
          Importer un jeu de données
        </Button>
      </div>
      <div className="my-2">
        {fragments.length === 0 && (
          <Alert type="info">{`Aucun jeu de donnée n'a a été importé`}</Alert>
        )}
        {fragments.length > 0 && (
          <DataGrid
            enableVirtualization={!isTested}
            className="border"
            columns={[
              {
                key: 'name',
                name: 'Colonne',
              },
              {
                key: 'source',
                name: 'Source',
              },
              {
                key: 'validity',
                name: 'Validité',
              },
              {
                key: 'dataset',
                name: 'Jeu de données',
              },
            ]}
            rows={rows}
          />
        )}
      </div>
    </div>
  );
};
