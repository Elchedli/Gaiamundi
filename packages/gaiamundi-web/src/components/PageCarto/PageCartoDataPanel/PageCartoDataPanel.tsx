import { PlusIcon } from '@heroicons/react/24/solid';
import { Alert } from 'components/Alert/Alert';
import { Button } from 'components/Button/Button';
import config from 'config';
import { useModal } from 'hooks/useModal';
import { ApiCollection } from 'interfaces/api';
import { Column } from 'interfaces/column';
import { DataFragment } from 'interfaces/data-fragment';
import Papa from 'papaparse';
import { FC, useMemo } from 'react';
import DataGrid from 'react-data-grid';
import { ConvertCSVToPageCartoData, idTables, testTables } from 'utils/parsing';
import { PageCartoDataForm } from './PageCartoDataForm';
type PageCartoPanelDataProps = {
  dataFragments: ApiCollection<DataFragment>;
  pageCartoId: number;
};

export const PageCartoPanelData: FC<PageCartoPanelDataProps> = ({
  dataFragments,
  pageCartoId,
}) => {
  const { showModal, hideModal } = useModal();
  const fragments = dataFragments?.data || [];
  const rows = useMemo(
    () =>
      fragments.reduce((acc, curr) => {
        const cols = curr.attributes.columns
          .filter((column) => {
            return !column.isGeoCode;
          })
          .map((column) => {
            return {
              ...column,
              dataset: curr.attributes.dataset.data.attributes.name,
            };
          });
        return acc.concat(cols);
      }, [] as (Column & { dataset: string })[]),
    [dataFragments]
  );
  // console.log('rows : ', rows);
  const fragmentsDatas = useMemo(() => {
    const csvDatas: string[][] = [];
    fragments.forEach(async (element) => {
      const urlCsvFile =
        config.API_URL +
        element.attributes.dataset.data.attributes.csv.data.attributes.url;
      Papa.parse(urlCsvFile, {
        download: true,
        complete: function (results: Papa.ParseResult<string>) {
          csvDatas.push(results.data);
        },
      });
    });
    const newTable: Array<any> = ConvertCSVToPageCartoData(
      testTables,
      idTables
    ).fuseObjectsUnique();
    return newTable;
  }, [dataFragments]);
  console.clear();
  console.log('fragments : ', fragments);
  console.log('Datas : ', fragmentsDatas);
  // console.log('fragments : ', fragments);
  // console.log('fragments : ', rows);
  return (
    <div>
      <div className="mt-5 text-right">
        <Button
          icon={PlusIcon}
          onClick={() =>
            showModal({
              title: 'Importer un jeu de données',
              Component: PageCartoDataForm,
              props: { pageCartoId, onSubmit: hideModal },
            })
          }
        >
          Importer un jeu de données
        </Button>
      </div>
      <div className="my-2">
        {fragments.length === 0 && (
          <Alert type="info">{`Aucun jeu de donnée n'a a été importé`}</Alert>
        )}
        {fragments.length > 0 && (
          <DataGrid
            className="border"
            columns={[
              {
                key: 'name',
                name: 'Colonne',
              },
              {
                key: 'source',
                name: 'Source',
              },
              {
                key: 'validity',
                name: 'Validité',
              },
              {
                key: 'dataset',
                name: 'Jeu de données',
              },
            ]}
            rows={rows}
          />
        )}
      </div>
    </div>
  );
};
