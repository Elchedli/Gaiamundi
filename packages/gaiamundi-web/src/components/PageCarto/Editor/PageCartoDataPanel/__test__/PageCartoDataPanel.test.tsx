import { render } from '@testing-library/react';
import { mockDataFragmentsApiCollection } from 'utils/mocks/data';
import { PageCartoPanelData } from '../PageCartoDataPanel';

describe('PageCartoPanelData', () => {
  it('should render the component', () => {
    const { getByText } = render(
      <PageCartoPanelData
        dataFragments={mockDataFragmentsApiCollection}
        pageCartoId={1}
      />
    );
    expect(getByText('Importer un jeu de donnÃ©es')).toBeInTheDocument();
  });

  it('should display an info alert if no data is present', () => {
    const { container, getByRole } = render(
      <PageCartoPanelData
        dataFragments={{
          data: [],
          meta: {
            pagination: {
              total: 0,
            },
          },
        }}
        pageCartoId={1}
      />
    );
    expect(getByRole('alert')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should display the data table if data is present', async () => {
    const { container, getByText, getAllByRole } = render(
      <PageCartoPanelData
        dataFragments={mockDataFragmentsApiCollection}
        pageCartoId={286}
      />
    );
    const datasetName =
      mockDataFragmentsApiCollection.data[0].attributes.dataset.data.attributes
        .name;
    const tabColumn = mockDataFragmentsApiCollection.data[0].attributes.columns;
    const tabHeader = getAllByRole('columnheader');
    expect(tabHeader).toHaveLength(4);
    tabColumn
      .filter((column) => !column.isGeoCode)
      .forEach((column) => {
        const row = getByText(column.name).parentNode;
        expect(row).toHaveTextContent(datasetName);
        expect(row).toHaveTextContent(column.source);
        expect(row).toHaveTextContent(column.validity);
      });
    expect(container).toMatchSnapshot();
  });
});
