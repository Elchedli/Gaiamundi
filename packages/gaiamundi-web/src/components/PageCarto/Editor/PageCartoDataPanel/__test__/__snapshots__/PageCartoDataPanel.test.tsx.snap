// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageCartoPanelData should display an info alert if no data is present 1`] = `
<div>
  <div
    class="p-3 my-3 rounded border border-blue-500 bg-blue-200 text-blue-700"
    role="alert"
  >
    <div
      class="flex flex-row"
    >
      <div>
        Aucun contenu à afficher.
      </div>
    </div>
  </div>
</div>
`;

exports[`PageCartoPanelData should display the data table if data is present 1`] = `
<div>
  <div>
    <div
      class="mt-5 text-right"
    >
      <button
        class="group inline-flex h-min w-fit items-center justify-center p-0.5 text-center font-medium focus:z-10 rounded-lg text-white bg-blue-700 border border-transparent hover:bg-blue-800 focus:ring-4 focus:ring-blue-300 disabled:hover:bg-blue-700 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800 dark:disabled:hover:bg-blue-600"
        data-testid="import-dataset"
        type="button"
      >
        <span
          class="flex items-center text-sm px-4 py-2 rounded-md !px-2"
        >
          <svg
            aria-hidden="true"
            class="w-5 h-5"
            fill="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clip-rule="evenodd"
              d="M12 3.75a.75.75 0 01.75.75v6.75h6.75a.75.75 0 010 1.5h-6.75v6.75a.75.75 0 01-1.5 0v-6.75H4.5a.75.75 0 010-1.5h6.75V4.5a.75.75 0 01.75-.75z"
              fill-rule="evenodd"
            />
          </svg>
          Importer un jeu de données
        </span>
      </button>
    </div>
    <div
      class="my-2"
    >
      <div
        aria-colcount="4"
        aria-rowcount="4"
        class="rdg r104f42s7-0-0-beta-22 border"
        dir="ltr"
        role="grid"
        style="grid-template-rows: 35px repeat(3, 35px); --rdg-header-row-height: 35px; --rdg-summary-row-height: 35px; --rdg-sign: 1; grid-template-columns: auto auto auto auto;"
      >
        <div
          aria-rowindex="1"
          class="rdg-header-row h197vzie7-0-0-beta-22"
          role="row"
          style="--rdg-grid-row-start: 1;"
        >
          <div
            aria-colindex="1"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="columnheader"
            style="grid-column-start: 1;"
            tabindex="0"
          >
            Colonne
          </div>
          <div
            aria-colindex="2"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="columnheader"
            style="grid-column-start: 2;"
            tabindex="-1"
          >
            Source
          </div>
          <div
            aria-colindex="3"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="columnheader"
            style="grid-column-start: 3;"
            tabindex="-1"
          >
            Validité
          </div>
          <div
            aria-colindex="4"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="columnheader"
            style="grid-column-start: 4;"
            tabindex="-1"
          >
            Jeu de données
          </div>
        </div>
        <div
          aria-rowindex="2"
          class="rdg-row r1otpg647-0-0-beta-22 rdg-row-even"
          role="row"
          style="--rdg-grid-row-start: 2; --rdg-row-height: 35px;"
        >
          <div
            aria-colindex="1"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 1;"
            tabindex="-1"
          >
            secondColumn
          </div>
          <div
            aria-colindex="2"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 2;"
            tabindex="-1"
          >
            source2
          </div>
          <div
            aria-colindex="3"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 3;"
            tabindex="-1"
          >
            2022
          </div>
          <div
            aria-colindex="4"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 4;"
            tabindex="-1"
          >
            firstDataset
          </div>
        </div>
        <div
          aria-rowindex="3"
          class="rdg-row r1otpg647-0-0-beta-22 rdg-row-odd"
          role="row"
          style="--rdg-grid-row-start: 3; --rdg-row-height: 35px;"
        >
          <div
            aria-colindex="1"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 1;"
            tabindex="-1"
          >
            thirdColumn
          </div>
          <div
            aria-colindex="2"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 2;"
            tabindex="-1"
          >
            source3
          </div>
          <div
            aria-colindex="3"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 3;"
            tabindex="-1"
          >
            2023
          </div>
          <div
            aria-colindex="4"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 4;"
            tabindex="-1"
          >
            firstDataset
          </div>
        </div>
        <div
          aria-rowindex="4"
          class="rdg-row r1otpg647-0-0-beta-22 rdg-row-even"
          role="row"
          style="--rdg-grid-row-start: 4; --rdg-row-height: 35px;"
        >
          <div
            aria-colindex="1"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 1;"
            tabindex="-1"
          >
            fourthColumn
          </div>
          <div
            aria-colindex="2"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 2;"
            tabindex="-1"
          >
            source4
          </div>
          <div
            aria-colindex="3"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 3;"
            tabindex="-1"
          >
            2024
          </div>
          <div
            aria-colindex="4"
            aria-readonly="true"
            aria-selected="false"
            class="rdg-cell c1wupbe7-0-0-beta-22"
            role="gridcell"
            style="grid-column-start: 4;"
            tabindex="-1"
          >
            firstDataset
          </div>
        </div>
        <div
          class="m1l09lto7-0-0-beta-22"
          data-measuring-cell-key="name"
          style="grid-column-start: 1; min-width: 80px;"
        />
        <div
          class="m1l09lto7-0-0-beta-22"
          data-measuring-cell-key="source"
          style="grid-column-start: 2; min-width: 80px;"
        />
        <div
          class="m1l09lto7-0-0-beta-22"
          data-measuring-cell-key="validity"
          style="grid-column-start: 3; min-width: 80px;"
        />
        <div
          class="m1l09lto7-0-0-beta-22"
          data-measuring-cell-key="dataset"
          style="grid-column-start: 4; min-width: 80px;"
        />
      </div>
    </div>
  </div>
</div>
`;
